django/db/models/query.py

```diff
@@ -136,6 +137,35 @@ def __iter__(self):
         return compiler.results_iter(tuple_expected=True)
 
 
+class NamedValuesListIterable(ValuesListIterable):
+    """
+    Iterable returned by QuerySet.values_list(named=True) that yields a
+    namedtuple for each row.
+    """
+
+    @staticmethod
+    @lru_cache()
+    def create_namedtuple_class(*names):
+        # namedtuple() is too slow to be called for every QuerySet evaluation
+        # so it's cached here.
+        return namedtuple('Row', names)
+
+    def __iter__(self):
+        queryset = self.queryset
+        if queryset._fields:
+            names = queryset._fields
+        else:
+            query = queryset.query
+            names = list(query.extra_select)
+            names.extend(query.values_select)
+            names.extend(query.annotation_select)
+        tuple_class = self.create_namedtuple_class(*names)
+        # tuple_class._make() is inlined here for better performance.
```

Comment:
Using `_make()`:
```
%timeit for x in Person.objects.values_list(named=True): pass
23.1 ms ± 438 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
Using inlined `_make()`:
```
%timeit for x in Person.objects.values_list(named=True): pass
18 ms ± 121 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
Comparing with plain tuples:
```
%timeit for x in Person.objects.values_list(): pass
14 ms ± 167 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
I think `tuple_class._make` refers to `tuple.__new__` but are the names intentionally different?

Reply:
No, I meant this `_make()`:
```
In [107]: namedtuple('Row', ('field',), verbose=True)                                              
from builtins import property as _property, tuple as _tuple                                        
from operator import itemgetter as _itemgetter   
from collections import OrderedDict              

class Row(tuple):                                
    'Row(field,)'                                

    __slots__ = ()                               

    _fields = ('field',)                         

    def __new__(_cls, field,):                   
        'Create new instance of Row(field,)'     
        return _tuple.__new__(_cls, (field,))    

    @classmethod                                 
    def _make(cls, iterable, new=tuple.__new__, len=len):                                          
        'Make a new Row object from a sequence or iterable'                                        
        result = new(cls, iterable)              
        if len(result) != 1:                     
            raise TypeError('Expected 1 arguments, got %d' % len(result))                          
        return result                            
...
```

I added this comment so it would be easier to understand from what I derived this.

Reply:
I'm still not sure what "tuple_class._make() is inlined here" means.

Reply:
Probably it's better to remove it in this case.